### PR TITLE
Refactor track design creation return message

### PR DESCRIPTION
**Pull request by spacek531**
This refactor implements non-fatal messages for saving track designs.

### CODE REVIEW

File: src/openrct2-ui/windows/Ride.cpp [hunk 1]
```diff
@@ -5363,11 +5363,15 @@ static void WindowRideMeasurementsDesignSave(rct_window* w)
...
     {
         auto errMessage = _trackDesign->CreateTrackDesignScenery(tds);
-        if (errMessage != STR_NONE)
+        if (!errMessage.Successful)
         {
-            ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage, {});
+            ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage.Message, {});
             return;
         }
+        if (errMessage.Message)
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp:5371]: This is not correct - this check will always be true. It needs to be compared against `STR_NONE`. This should preferably be added in the form of a method called `HasMessage()` that does this check.

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -998,11 +998,15 @@ std::unique_ptr<TrackDesign> Ride::SaveToTrackDesign(TrackDesignState& tds) cons
...
     auto td = std::make_unique<TrackDesign>();
     auto errMessage = td->CreateTrackDesign(tds, *this);
-    if (errMessage != STR_NONE)
+    if (!errMessage.Successful)
     {
-        ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage, {});
+        ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage.Message, {});
         return nullptr;
     }
+    if (errMessage.Message)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1006]: See previous.

**Comment by Gymnasiast** [comment]
Nice refactor. This is the kind of thing that `ResultWithMessage` was created for.

I take it you plan to make changes that actually return `true` with a message in a future PR?

**Comment by spacek531** [comment]
That is correct. I will use it in speedy block brakes - saving a design with non-default block brake speed set is a nonfatal warning.

File: src/openrct2-ui/windows/Ride.cpp (hunk 1 above)
Comment (spacek531) [reply src/openrct2-ui/windows/Ride.cpp:5371]: Ah I had missed that part of ResultWithMessage class definition

**Comment by spacek531** [comment]
HasMessage() is now implemented

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**